projects/fsastorefrontlib/src/cms-components/sales-illustration/components/sales-illustration-main/sales-illustration-main.component.ts

```diff
@@ -0,0 +1,272 @@
+    private activatedRoute: ActivatedRoute,
+    private productService: FSProductService,
+    private formDataStorageService: FormDataStorageService,
+    private formDataService: FormDataService,
+    private pricingService: PricingService,
+    private chartService: ChartService,
+    private cartService: FSCartService,
+    private routingService: RoutingService,
+    private breakpointService: BreakpointService,
+    public location: Location
+  ) {}
+
+  ngAfterViewInit(): void {
+    combineLatest([this.breakpointService.breakpoint$, this.nav.changes])
+      .pipe(
+        tap(([breakpoint, change]) => {
+          breakpoint === BREAKPOINT.xs ? change.first.select(2) : null;
+        }),
+        takeUntil(this.destroyed$)
+      )
+      .subscribe();
+  }
+
+  ngOnInit(): void {
+    fromEvent(window, 'resize')
+      .pipe(
+        debounceTime(300),
+        tap(e => this.resizeChart()),
+        takeUntil(this.destroyed$)
+      )
+      .subscribe();
+
+    this.productCode$ = this.activatedRoute.params.pipe(
+      map(param => param.savingsProductCode),
+      logger('Product Code'),
+      shareReplay()
+    );
+
+    this.pricingData$ = this.getPricingData();
+
+    this.product$ = combineLatest([this.productCode$, this.pricingData$]).pipe(
+      filter(
+        ([_, pricingData]) => pricingData.priceAttributeGroups !== undefined
+      ),
+      switchMap(([productCode, pricingData]) =>
+        this.productService.getCalculatedProductData(productCode, pricingData)
+      ),
+      filter(product => !!product),
+      map((product: any) => this.formatProduct(product)),
+      logger('Changed Product'),
+      shareReplay()
+    );
+
+    this.chartOptions$ = this.product$.pipe(
+      switchMap((product: any) =>
+        this.chartService.getChartOptions(product.salesIllustrationDiagramData)
+      )
+    );
+
+    this.tableData$ = this.pagination$.pipe(
+      withLatestFrom(this.product$),
+      map(([pagination, product]) => {
+        const {
+          contributionSeries,
+          expectedSavingsSeries,
+          interestSeries,
+          years,
+        } = product.salesIllustrationDiagramData;
+
+        return {
+          contributionSeries: PaginationHelper.getPaginationResults(
+            pagination,
+            contributionSeries
+          ),
+          expectedSavingsSeries: PaginationHelper.getPaginationResults(
+            pagination,
+            expectedSavingsSeries
+          ),
+          interestSeries: PaginationHelper.getPaginationResults(
+            pagination,
+            interestSeries
+          ),
+          years: PaginationHelper.getPaginationResults(pagination, years),
+        };
+      })
+    );
+  }
+
+  onChartInit(ec: ECharts) {
+    this.eChartsInstance = ec;
+  }
+
+  resizeChart() {
+    if (this.eChartsInstance) {
+      this.eChartsInstance.resize();
+    }
+  }
+
+  onPageChange(currentPage: number) {
+    this.pagination$.next({
+      ...this.pagination$.value,
```

Comment:
We should **not** try to get/set the value via `.value`. This basically breaks down the stream...
More on this can be found [here](https://stackoverflow.com/questions/62262008/rxjs-behaviorsubject-proper-use-of-value) and [here](https://stackoverflow.com/questions/37089977/how-to-get-current-value-of-rxjs-subject-or-observable/45227115#45227115)